### PR TITLE
Fix section navigation

### LINKED ISSUE #106: Respect order number in navigation
The contributor has the option to add an order number for the page, which should influence its order of display in the navigation.

This is not currently reflecting since the navigation redesign

### PR DESCRIPTION
**Pull request by yndajas**
Closes #106

- Fix top-level section sorting
- Use more transparent names for variables and templates
- Maintenance

### CODE REVIEW

File: src/_includes/site_navigation.html [hunk 1]
```diff
@@ -9,21 +9,21 @@
         {% for section in sections_by_title %}
           <li class="related__section-link">
```
Comment (lookupdaily) [inline src/_includes/site_navigation.html:10]: Is now the time to rename the sass classes too? Perhaps to `site-navigation` and `site-navigation__`

**Review by lookupdaily — APPROVED** [review]
Thanks for doing this! 

I keep wanting to rename the css classes too, so have left a comment re that - but equally happy to pick up in a separate PR :)

File: src/_includes/site_navigation.html (hunk 1 above)
Comment (yndajas) [reply src/_includes/site_navigation.html]: Yep, good idea! I've updated the relevant commit - do you think the new names make sense?

Comment (yndajas) [reply src/_includes/site_navigation.html]: (Also changed the `related_order` property to `navigation_order`

Comment (lookupdaily) [reply src/_includes/site_navigation.html]: Looks perfect! Yes and navigation_order makes sense to me too :)

**Pull request merged**